lib/webdrivers/chromedriver.rb

```diff
@@ -73,18 +80,56 @@ def chrome_version
       end
 
       def chrome_on_windows
-        reg = 'HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\chrome.exe'
-        ps  = "(Get-Item (Get-ItemProperty '#{reg}').'(Default)').VersionInfo.ProductVersion"
-        `powershell #{ps}`
+        if browser_defined?
+          Webdrivers.logger.debug "Browser executable: '#{browser_exe}'"
+          return `powershell (Get-ItemProperty '#{browser_exe}').VersionInfo.ProductVersion`
+        end
+
+        # Default to Google Chrome
+        reg        = 'HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\chrome.exe'
+        executable = `powershell (Get-ItemProperty '#{reg}' -Name '(default)').'(default)'`.strip
+        Webdrivers.logger.debug "Browser executable: '#{executable}'"
+        ps = "(Get-Item (Get-ItemProperty '#{reg}').'(default)').VersionInfo.ProductVersion"
+        `powershell #{ps}`.strip
       end
 
       def chrome_on_linux
-        `google-chrome --version`
+        if browser_defined?
+          Webdrivers.logger.debug "Browser executable: '#{browser_exe}'"
+          return `#{browser_exe} --product-version`.strip if browser_defined?
+        end
+
+        # Default to Google Chrome
+        executable = `which google-chrome`.strip
+        Webdrivers.logger.debug "Browser executable: '#{executable}'"
+        `#{executable} --product-version`.strip
       end
 
       def chrome_on_mac
-        loc = Shellwords.escape '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome'
-        `#{loc} --version`
+        if browser_defined?
+          Webdrivers.logger.debug "Browser executable: '#{browser_exe}'"
+          return `#{browser_exe} --version`.strip if browser_defined?
+        end
+
+        # Default to Google Chrome
+        executable = Shellwords.escape '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome'
+        Webdrivers.logger.debug "Browser executable: #{executable}"
+        `#{executable} --version`.strip
+      end
+
+      #
+      # Returns true if user defines path to the browser executable.
+      #
+      def browser_defined?
```

Comment:
I think we should just use `!browser_binary` instead of defining a new helper method for this? I think the method is clear enough by itself without making the class longer?

Reply:
Makes sense. Updated!